src/app/betting/feature/place-bet-page.component.ts

```diff
@@ -219,8 +361,51 @@ export default class PlaceBetPageComponent {
     });
   }
 
-  // When player list changes (e.g. someone's score changes externally)
-  // clear selected opponent so it doesn't get orphaned
+  generateBetTypeObject(type: BetType) {
+    switch (type) {
+      case BetType.DuelWinner:
+        return { betType: type, betTypeString: 'Duel Winner' };
+      default:
+        return { betType: BetType.Manual, betTypeString: 'Manual' };
+    }
+  }
+
+  private generateDetails(): Json {
```

Comment:
See comment in #17

Reply:
I'll plan on fixing this on the other branch so that I don't have to deal with a merge